testing/database.rst

```diff
@@ -4,40 +4,127 @@
 How to Test Code that Interacts with the Database
 =================================================
 
-If your code interacts with the database, e.g. reads data from or stores data
-into it, you need to adjust your tests to take this into account. There are
-many ways to deal with this. In a unit test, you can create a mock for
-a ``Repository`` and use it to return expected objects. In a functional test,
-you may need to prepare a test database with predefined values to ensure that
-your test always has the same data to work with.
+Configuring a Database for Tests
+--------------------------------
 
-.. note::
+Tests that interact with the database should use their own separate database to
+not mess with the databases used in the other :ref:`configuration environments <configuration-environments>`.
+To do that, edit or create the ``.env.test.local`` file at the root directory of
+your project and define the new value for the ``DATABASE_URL`` env var:
 
-    If you want to test your queries directly, see :doc:`/testing/doctrine`.
+.. code-block:: bash
+
+    # .env.test.local
+    DATABASE_URL=mysql://USERNAME:PASSWORD@127.0.0.1/DB_NAME
 
 .. tip::
 
-    A popular technique to improve the performance of tests that interact with
-    the database is to begin a transaction before every test and roll it back
-    after the test has finished. This makes it unnecessary to recreate the
-    database or reload fixtures before every test. A community bundle called
-    `DoctrineTestBundle`_ provides this feature.
+    A common practice is to append the ``_test`` suffix to the original database
+    names in tests. If the database name in production is called ``project_acme``
+    the name of the testing database could be ``project_acme_test``.
 
-Mocking the ``Repository`` in a Unit Test
------------------------------------------
+The above assumes that each developer/machine uses a different database for the
+tests. If the entire team uses the same settings for tests, edit or create the
+``.env.test`` file instead and commit it to the shared repository. Learn more
+about :ref:`using multiple .env files in Symfony applications <configuration-multiple-env-files>`.
 
-If you want to test code which depends on a Doctrine repository in isolation,
-you need to mock the ``Repository``. Normally you inject the ``EntityManager``
-into your class and use it to get the repository. This makes things a little
-more difficult as you need to mock both the ``EntityManager`` and your repository
-class.
+Resetting the Database Automatically Before each Test
+-----------------------------------------------------
 
-.. tip::
+Tests should be independent from each other to avoid side effects. However, if
+some test modifies the database (for example by adding or removing an element)
+it could impact the other tests. Run the following command to install a bundle
+that ensures that each test is run with a new and unmodified copy of the database:
```

Comment:
This is not correct. It isolates queries, but they are still run in same database


Something like this perhaps would be more accurate
```suggestion
that ensures that SQL queries of each test are isolated in their own sandbox
```

2 instances of this

Reply:
indeed tests are not run within their own copy of a database

Reply:
@ostrolucky you are right. I've reworded this a bit to explain things better and more precisely. Thanks!

Reply:
cool, thank you too